### ISSUE TITLE
Trying to build for a language that does not exist leads to errors

### ISSUE THREAD

**Issue by sknorr** [openSUSE/docserv#262]
Toms tried this:

```
dscmd trigger --products=sles --docset=15-SP3 --langs=en --target=test
```

Despite the `en` language not being defined or accepted, Ds2 tried to build navigation:

```
2021-07-06 11:33:36,267 - DEBUG - No buildable documentation for sles/15-SP3/en. Will update navigation only.
```

This then later manifested in a traceback.

This (presumably) broke as a consequence of https://github.com/openSUSE/docserv/commit/1fcb82a0ec8c3db8b346e042d77c61ba2068b367

We really need to make sure that if there's no buildable documentation, the requested language must [at least] be mentioned in the list of ui_languages.

**Comment by sknorr** [comment]
There's a bit more to fix here, I think:

```
2021-07-06 11:33:36,164 - INFO - Queueing {"target": "test", "product": "sles", "docset": "15-SP3", "lang": "en", "id": "a3c7866bc"}
127.0.0.1 - - [06/Jul/2021 11:33:36] "POST / HTTP/1.1" 200 -
2021-07-06 11:33:36,176 - DEBUG - Valid build instruction: a3c7866bc
2021-07-06 11:33:36,176 - DEBUG - Stitching XML config directory to /tmp/docserv_stitch__2hkikdu/productconfig_simplified_test.xml
2021-07-06 11:33:36,176 - DEBUG - Stitching command: [repodir]/bin/docserv-stitch --simplify --revalidate-only --valid-languages="ar-ar cs-cz de-de en-us es-es fi-fi fr-fr hu-hu it-it ja-jp ko-kr nl-nl pl-pl pt-br ru-ru sv-se zh-cn zh-tw" [configdir]/config.d /tmp/docserv_stitch__2hkikdu/productconfig_simplified_test.xml
2021-07-06 11:33:36,262 - DEBUG - Stitching of [configdir]/config.d successful
2021-07-06 11:33:36,267 - DEBUG - No buildable documentation for sles/15-SP3/en. Will update navigation only.
2021-07-06 11:33:36,267 - INFO - Finished build instruction a3c7866bc
2021-07-06 11:33:36,267 - DEBUG - Cleaning up after "a3c7866bc" (success)
2021-07-06 11:33:36,267 - DEBUG - Cleaning up a3c7866bc, rm -rf [configdir]/0serve/en/sles/15-SP3
2021-07-06 11:33:36,270 - DEBUG - Cleaning up a3c7866bc, docserv-build-navigation  --product="sles" --docset="15-SP3" --stitched-config="/tmp/docserv_stitch__2hkikdu/productconfig_simplified_test.xml" --ui-languages="en-us de-de fr-fr" --default-ui-language="en-us"  --cache-dir="[configdir]/0cache/docserv-local-test/test" --template-dir="[configdir]/template-docserv" --output-dir="/tmp/docserv_navigation_dj98z1oo" --base-path="/"
2021-07-06 11:33:36,282 - WARNING - Clean up failed! Unexpected return value 1 for 'docserv-build-navigation  --product="sles" --docset="15-SP3" --stitched-config="/tmp/docserv_stitch__2hkikdu/productconfig_simplified_test.xml" --ui-languages="en-us de-de fr-fr" --default-ui-language="en-us"  --cache-dir="[configdir]/0cache/docserv-local-test/test" --template-dir="[configdir]/template-docserv" --output-dir="/tmp/docserv_navigation_dj98z1oo" --base-path="/" '
2021-07-06 11:33:36,283 - DEBUG - Writing message to [configdir]/0cache/docserv-message-2021-07-06T11:33:36+02:00-90pifueu.txt
2021-07-06 11:33:36,283 - DEBUG - Cleaning up a3c7866bc, rsync -r /etc/docserv/server-root-files-docserv/ /tmp/docserv_navigation_dj98z1oo
2021-07-06 11:33:36,388 - WARNING - Clean up failed! Unexpected return value 23 for 'rsync -r /etc/docserv/server-root-files-docserv/ /tmp/docserv_navigation_dj98z1oo'
2021-07-06 11:33:36,389 - DEBUG - Writing message to [configdir]/0cache/docserv-message-2021-07-06T11:33:36+02:00-vooi5y2t.txt
2021-07-06 11:33:36,390 - DEBUG - Cleaning up a3c7866bc, rsync -lr /tmp/docserv_navigation_dj98z1oo/ [configdir]/0serve
2021-07-06 11:33:36,405 - DEBUG - Cleaning up a3c7866bc, rm -rf /tmp/docserv_navigation_dj98z1oo
2021-07-06 11:33:36,413 - DEBUG - Cleaning up a3c7866bc, rm -rf /tmp/docserv_deliverable_sles_15-SP3_en_eoydk9ay
2021-07-06 11:33:36,420 - DEBUG - Cleaning up a3c7866bc, rm -rf [configdir]/0branches/CGMTA4BZHMR4
```

(from Toms's pastebin)

**Comment by sknorr** [comment]
Fixed with 51b102b, 8124856, caecfca.
